overviewer_core/src/primitives/cave.c

```diff
@@ -78,15 +78,15 @@ cave_hidden(void* data, RenderState* state, int32_t x, int32_t y, int32_t z) {
      * at this point of the code the block has no skylight
      * but a deep sea can be completely dark
      */
-    if ((getArrayShort3D(state->blocks, x, y, z) == 9) ||
-        (get_data(state, BLOCKS, x, y + 1, z) == 9)) {
 
+    if ((getArrayShort3D(state->blocks, x, y, z) == 9) || (getArrayShort3D(state->blocks, x, y, z) == 8) ||
+        (get_data(state, BLOCKS, x, y + 1, z) == 9) || (get_data(state, BLOCKS, x, y + 1, z) == 8)) {
```

Comment:
Is the compiler smart enough to optimise away those repeated function calls?

Reply:
No idea, not a C programmer ;).  Want them in temporary variables?

Reply:
> Is the compiler smart enough to optimise away those repeated function calls?
Best answer I can give to that without checking a compiled version is "maybe".
I'd probably err on the side of a (clearly named) temporary variable either way.

Reply:
ok will do